src/simplify_algebra.cpp

```diff
@@ -453,6 +491,23 @@ struct find_splits
             if(op.name() == "slice")
                 continue;
 
+            if(contains(op.name(), "reduce"))
+            {
+                auto split_front = splits.front();
+                auto slc         = any_cast<op::slice>(split_front->get_operator());
+                auto slc_axes    = slc.axes;
+
+                auto reduce_axes = get_reduce_axes(start);
+                // axes of slice and reduce op cannot have overlap
+                if(std::any_of(slc_axes.begin(), slc_axes.end(), [&](auto axis) {
+                       return (std::find(reduce_axes.begin(), reduce_axes.end(), axis) !=
+                               reduce_axes.end());
+                   }))
+                {
+                    continue;
+                }
+            }
```

Comment:
We need and else here to check that the operator is pointwise. If its not reduce or pointwise then we should `continue`.